### PR TITLE
Challenge/min and max detector

### LINKED ISSUE #11: Max and Min Detector
| **Field**        | **Details**                                                                                                         |
|------------------|---------------------------------------------------------------------------------------------------------------------|
| **Name**         | New Challenge                                                                                                       |
| **Description**  | Write a program that finds the maximum and minimum values in a list of numbers.                                                           |
| **Title**        | Max and Min Detector                                                                                               |

### PR DESCRIPTION
**Pull request by tosan-okome**
Wrote and tested a function to calculate the min and max of a given list of numbers. 

Resolves #11 

Code review checklist below: 

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [ ] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations
- [x] The function is not called in the function file

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/find_min_max.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+A module for generating the min and max of a list of numbers.
```
Comment (Safiya-hash) [inline solutions/find_min_max.py]: Maybe this?
```suggestion
A module for finding the min and max of a list of numbers.
```

File: solutions/find_min_max.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Module contents:
+    > find_min_max: generates the min and max of a list of numbers.
```
Comment (Safiya-hash) [inline solutions/find_min_max.py]: Maybe changing the word generates to finding will be easier to understand.
```suggestion
    > find_min_max: generates the min and max of a list of numbers.
```

File: solutions/find_min_max.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        >>> find_min_max([9.7, 67.9, 8, 9.2])
+        (8, 67.9)
+    """
+    if not numbers:
+        raise ValueError("The list cannot be empty.")
+
+    if not all(isinstance(num, (int, float)) for num in numbers):
+        raise ValueError("All elements in the list must be numbers.")
+
+    return min(numbers), max(numbers)
```
Comment (Safiya-hash) [inline solutions/find_min_max.py:43]: What you think of writing self-generated min max algorithms (using strategies instead of python functions) like comparing each number of lists then return the max/min? 😃

**Review by Safiya-hash — COMMENTED** [review]
The code looks amazing and clean with clear documentation. I just added few points to consider, let me know what you think. 🤗

File: solutions/find_min_max.py (hunk 2 above)
Comment (tosan-okome) [reply solutions/find_min_max.py]: You're right! It maintains consistency with the function name.

File: solutions/find_min_max.py (hunk 3 above)
Comment (tosan-okome) [reply solutions/find_min_max.py:43]: Nice suggestion. 
I opted for a function because I was trying to keep the code as easy and readable as possible. A strategy I could use is sorting the list in ascending order and returning the first and last value in the sorted list. But that would have still required me to use a function (the "sort" function).  😄

File: solutions/find_min_max.py (hunk 2 above)
Comment (tosan-okome) [reply solutions/find_min_max.py]: Fixed this!

File: solutions/find_min_max.py (hunk 3 above)
Comment (Safiya-hash) [reply solutions/find_min_max.py:43]: Okay, it was totally read and understandable. 
Thanks for your work. 🌺

**Review by Safiya-hash — APPROVED** [review]

**Comment by Safiya-hash** [comment]
As everything was in order I approved and merged the pull request, good luck. 🚀
